### DISCUSSION TITLE
Can't train correctly a tesseract model

### DISCUSSION THREAD

**Discussion by igonro** [buliasz/tesstrain-windows-gui#35] [Q&A]
Hello, I'm new at training in Tesseract, so I will explain a bit my problem in order to get a better understanding of what I am doing wrong.

My problem is simple, I want to use Tesseract to read text in images. Sadly the font is a pixel font called [Munro](https://www.wfonts.com/font/munro), so the predictions using default Tesseract config are very poor.

Then I started to read that it is possible to train a Tesseract model for a specific font. I generated a dataset of 1000 random sentences (~8-10 words per sentence) using [text2image](https://github.com/tesseract-ocr/tesseract/blob/main/doc/text2image.1.asc) and following a script similar to [this one](https://github.com/astutejoe/tesseract_tutorial/blob/main/split_training_text.py).

I trained using your GUI app (which is awesome by the way) without changing too much the parameters (10000 iterations, 0.002 LR, without starting model...). The BCER at the end of the training was 0.30% (or something like that), but when I tried to predict some images of the groundind truth the prediction was something like "BBBBBB".

What I am doing wrong? I need to train more iterations? Or maybe more data?

If you need more information, ask me.

Thank you very much!

**Comment by igonro** [comment]
Maybe it could be useful (for noobs like me) that this repo would contain an utility script for creating the `X-ground-truth` folder from a given font. Because the only thing I wanted to do was select the font file, train the model, and use it.

**Reply by buliasz** [reply]
Hi @igonro ,
It is an interesting use case. I think what I'd personally do is I'd use some starting model anyway, any for the language you want to recognize.
But from what you describe I think you just need more training. I think 1000 sentences should be enough to get quite good results. At least I've tried with less and got very good results. Instead of iterations you may want to use epochs (negative value, e.g. -10) just for convenience, and set the Target error rate to 0 (just to finish when all the epochs are done). Next time you train use the previously trained model as a start model.

There may be though some other limitation or detail which I'm not aware of. Maybe the main Tesseract repository developers would know better. You may want to ask there if the above won't help.

**Reply by buliasz** [reply]
![settings_proposition](https://user-images.githubusercontent.com/22753806/204439772-a35ee2e0-5c87-4df3-931a-874bb77d5922.png)
Try something like this

**Reply by igonro** [reply]
Thank you so much! I will try it as soon as I can.

**Comment by Blu3train** [comment]
Since I've never used epochs, does the value of epochs equal as interactions the number of sentences multiplied by epochs?
Thank you

**Reply by buliasz** [reply]
I don't know how exactly it is implemented in the Tesseract tools but normally in ML software "An epoch is when all the training data is used at once and is defined as the total number of iterations of all the training data in one cycle for training the machine learning model" ([link](https://www.simplilearn.com/tutorials/machine-learning-tutorial/what-is-epoch-in-machine-learning)).
